plugin/mlsdbprovider.cpp

```diff
@@ -269,78 +272,106 @@ void MlsdbProvider::calculatePositionAndEmitLocation()
     qCDebug(lcGeoclueMlsdbPosition) << "have" << m_cellWatcher->cells().size() << "neighbouring cells";
     QList<CellPositioningData> cells;
     quint32 maxNeighborSignalStrength = 1;
-    QSet<quint32> seenCellIds;
+    QSet<quint64> seenCellIds;
     Q_FOREACH (const QSharedPointer<QOfonoExtCell> &c, m_cellWatcher->cells()) {
         CellPositioningData cell;
+        MlsdbCellType cellType = MLSDB_CELL_TYPE_LTE;
+        quint32 locationCode = 0;
+        quint32 cellId = 0;
         if (c->cid() != -1 && c->cid() != 0) {
             // gsm / wcdma
-            cell.cellId = c->cid();
+            cellType = MLSDB_CELL_TYPE_GSM_WCDMA;
```

Comment:
I have to check the value of cid() / ci() anyway, since we often get invalid values like "ignoring neighbour cell with no cell id: 3 -1 -1 -1 -1" reported (i.e., that one's an LTE cell with invalid values for the various fields), so implicitly detecting the type doesn't cost us anything but saves us one needless check.